src/Service.Host/client/src/components/parts/PartTemplate.js

```diff
@@ -0,0 +1,520 @@
+                  hasDataSheet: 'N',
+                  hasNumberSequence: '',
+                  nextNumber: '',
+                  allowVariants: 'N',
+                  variants: '',
+                  accountingCompany: '',
+                  productCode: '',
+                  stockControlled: '',
+                  linnProduced: '',
+                  rmfgCode: '',
+                  bomType: '',
+                  assemblyTechnology: '',
+                  allowPartCreation: '',
+                  paretoCode: ''
+              }
+            : null
+    );
+    const [prevPartTemplate, setPrevPartTemplate] = useState(
+        creating()
+            ? {
+                  partRoot: '',
+                  description: '',
+                  hasDataSheet: 'N',
+                  hasNumberSequence: '',
+                  nextNumber: '',
+                  allowVariants: 'N',
+                  variants: '',
+                  accountingCompany: '',
+                  productCode: '',
+                  stockControlled: '',
+                  linnProduced: '',
+                  rmfgCode: '',
+                  bomType: '',
+                  assemblyTechnology: '',
+                  allowPartCreation: '',
+                  paretoCode: ''
+              }
+            : null
+    );
+
+    useEffect(() => {
+        if (item && item !== prevPartTemplate) {
+            setPartTemplate(item);
+            setPrevPartTemplate(item);
+        }
+    }, [item, prevPartTemplate]);
+
+    const useStyles = makeStyles(() => ({
+        thinPage: {
+            width: '60%',
+            margin: '0 auto'
+        }
+    }));
+    const classes = useStyles();
+
+    const handleSaveClick = () => {
+        if (creating()) {
+            addItem(partTemplate);
+        } else {
+            updateItem(itemId, partTemplate);
+        }
+        setEditStatus('view');
+    };
+
+    const saveEnabled = () => {
+        return (
+            !partTemplate.partRoot ||
+            !partTemplate.description ||
+            !partTemplate.hasDataSheet ||
+            !partTemplate.allowVariants ||
+            !partTemplate.allowPartCreation ||
+            !partTemplate.hasNumberSequence
+        );
+    };
+
+    const handleCancelClick = () => {
+        if (creating()) {
+            setPartTemplate(prevPartTemplate);
+        } else {
+            setPartTemplate(item);
+            setEditStatus('view');
+        }
+    };
+
+    const handleFieldChange = (propertyName, newValue) => {
+        if (viewing() || creating()) {
+            setEditStatus('edit');
+        }
+
+        setPartTemplate(x => ({ ...x, [propertyName]: newValue }));
+    };
+
+    const handleProductCodeChange = product => {
+        handleFieldChange('productCode', product.name);
+        handleFieldChange('productAnalysisCodeDescription', product.description);
+    };
+
+    const allowedToEdit = () => {
+        return privileges?.some(priv => priv === 'part.admin');
```

Comment:
I thought we were replacing these front end privilege checks with Hypermedia links